### PR TITLE
Refactor extension attribute addition for modules and types.

### PR DESCRIPTION
**Pull request by nojaf**
## Description

Fixes the problem discovered in https://github.com/dotnet/fsharp/pull/16291.
//cc @edgarfgp 

## Checklist

- [x] Test cases added
- [ ] Performance benchmarks added in case of performance changes
- [ ] Release notes entry updated:
    > Please make sure to add an entry with short succint description of the change as well as link to this pull request to the respective release notes file, if applicable.
    >
    > Release notes files:
    > - If anything under `src/Compiler` has been changed, please make sure to make an entry in `docs/release-notes/FSharp.Compiler.Service/<version>.md`, where `<version>` is usually "highest" one, e.g. `42.8.200`
    > - If language feature was added (i.e. `LanguageFeatures.fsi` was changed), please add it to `docs/releae-notes/Language/preview.md`
    > - If a change to `FSharp.Core` was made, please make sure to edit `docs/release-notes/FSharp.Core/<version>.md` where version is "highest" one, e.g. `8.0.200`.

    > Examples of release notes entries:
    > - Respect line limit in quick info popup - https://github.com/dotnet/fsharp/pull/16208
    > - More inlines for Result module - https://github.com/dotnet/fsharp/pull/16106
    > - Miscellaneous fixes to parens analysis - https://github.com/dotnet/fsharp/pull/16262
    >

    > **If you believe that release notes are not necessary for this PR, please add `NO_RELEASE_NOTES` label to the pull request.**

### CODE REVIEW

**Comment by nojaf** [comment]
Ready for review

File: src/Compiler/TypedTree/TypedTreeOps.fs [hunk 1]
```diff
@@ -10571,23 +10571,42 @@ let (|EmptyModuleOrNamespaces|_|) (moduleOrNamespaceContents: ModuleOrNamespaceC
             None
     | _ -> None
 
-let tryAddExtensionAttributeIfNotAlreadyPresent
+let tryFindExtensionAttribute (attribs: Attrib list): Attrib option =
+     List.tryFind
+         (fun (a: Attrib) ->
+            a.TyconRef.CompiledRepresentationForNamedType.BasicQualifiedName = "System.Runtime.CompilerServices.ExtensionAttribute")
+         attribs
```
Comment (vzarytovskii) [inline src/Compiler/TypedTree/TypedTreeOps.fs:10578]: I am wondering if this should go via usual mechanism in TcGlobals (where we do `findSysAttrib`/`findOrCreateSysAttrib` and then use TryFindILAttribute/TryFindFSharpAttribute)? The same way we do for `IsReadOnlyAttribute` for example.

Comment (T-Gro) [reply src/Compiler/TypedTree/TypedTreeOps.fs:10578]: `TcGlobals` already has this:

`  member val attrib_ExtensionAttribute = findSysAttrib "System.Runtime.CompilerServices.ExtensionAttribute"`

**Review by T-Gro — APPROVED** [review]

File: src/Compiler/TypedTree/TypedTreeOps.fs (hunk 1 above)
Comment (nojaf) [reply src/Compiler/TypedTree/TypedTreeOps.fs]: Thanks, for pointing this out. I've tried to address this.
Let me know if it is more or less what you had in mind.

Comment (T-Gro) [reply src/Compiler/TypedTree/TypedTreeOps.fs]: This is good, thanks for the update.

**Comment by edgarfgp** [comment]
Just checked in the recently released `8.0.100` and it was not there :(. @vzarytovskii  Any idea if this will make to the next servicing release ?

**Comment by vzarytovskii** [comment]
> Just checked in the recently released `8.0.100` and it was not there :(. @vzarytovskii  Any idea if this will make to the next servicing release ? 

You mean 8.0.101? It's a servicing release. This didn't hit the bar for servicing release. It will be in 8.0.200. You should try in nightlies.

**Pull request merged**